portfolio/src/main/webapp/script.js

```diff
@@ -305,31 +403,37 @@ function continentsCheckBoxListener(countriesObj) {
 
   document.getElementById("na_checkbox").addEventListener("change", function() {
     countriesObj.remakeJsonByContinent(this.checked);
+    countriesObj.checkBoxSkipToggle = true;
     document.getElementById("skip_btn").click();
   });
 
   document.getElementById("sa_checkbox").addEventListener("change", function() {
     countriesObj.remakeJsonByContinent(undefined, this.checked);
+    countriesObj.checkBoxSkipToggle = true;
     document.getElementById("skip_btn").click();
   });
 
   document.getElementById("eu_checkbox").addEventListener("change", function() {
     countriesObj.remakeJsonByContinent(undefined, undefined, this.checked);
+    countriesObj.checkBoxSkipToggle = true;
     document.getElementById("skip_btn").click();
   });
 
   document.getElementById("af_checkbox").addEventListener("change", function() {
     countriesObj.remakeJsonByContinent(undefined, undefined, undefined, this.checked);
+    countriesObj.checkBoxSkipToggle = true;
     document.getElementById("skip_btn").click();
   });
 
   document.getElementById("as_checkbox").addEventListener("change", function() {
     countriesObj.remakeJsonByContinent(undefined, undefined, undefined, undefined, this.checked);
+    countriesObj.checkBoxSkipToggle = true;
     document.getElementById("skip_btn").click();
   });
 
   document.getElementById("oc_checkbox").addEventListener("change", function() {
     countriesObj.remakeJsonByContinent(undefined, undefined, undefined, undefined, undefined, this.checked);
+    countriesObj.checkBoxSkipToggle = true;
     document.getElementById("skip_btn").click();
   });
```

Comment:
Since this code to add event listeners to the checkboxes is repeated a bunch of times, it probably makes sense to add a class to each checkbox, select them all with the class, and loop over them to create the listeners. You could also look into event delegation to handle any checkbox click with a single listener. Don't worry about making the change for this PR but keep it in the refactor backlog.